git-compat-util.h

```diff
@@ -127,7 +127,9 @@
 /* Approximation of the length of the decimal representation of this type. */
 #define decimal_length(x)	((int)(sizeof(x) * 2.56 + 0.5) + 1)
 
-#if defined(__sun__)
+#ifdef __MINGW64__
+#define _POSIX_C_SOURCE 1
```

Comment:
I wonder how much of a fall-out this has on things other than `*time_r()`...

Reply:
I chose 1 to minimize the fall-out, but I can only see it as an advantage for compatibility with a codebase that targets POSIX anyway.

The only possible undesired effect I was able to track was its interaction with __USE_MINGW_ANSI_STDIO, but that not a problem unless the settings for it in the Makefile/cmake change as well AFAIK.

There is also the fact, this has been in use in the windows fork for a while without any known sideffects.

But I could change it to use the less aggresive _POSIX_THREAD_SAFE_FUNCTIONS which is mentioned in the list and which would make this even safer

Reply:
Alternatively we could revert this change and go with the one Mike propossed on list instead, which will also make the MSVC and gcc builds more similar (both using the fallback), and which might be a good change to do early in this release cycle as well.

Reply:
> There is also the fact, this has been in use in the windows fork for a while without any known sideffects.

Oh, so this has withstood the test of time in Git for Windows? _goes and checks_ ... indeed, it is in git-for-windows/git@adf0cd8d14262e56db39573287a0c16555eb892b.

That information would have been quite good in the cover letter or even in the commit message.